lib/money/money.rb

```diff
@@ -266,7 +271,11 @@ def fraction(rate)
   #     #=> [#<Money value:20.00 currency:USD>, #<Money value:10.00 currency:USD>]
   #   Money.new(30, "USD").allocate([Rational(2, 3), Rational(1, 3)])
   #     #=> [#<Money value:20.00 currency:USD>, #<Money value:10.00 currency:USD>]
-  def allocate(splits)
+
+  # @example left over pennies distributed reverse order when using roundrobin_reverse strategy
+  #   Money.new(10.01, "USD").allocate([0.5, 0.5], :roundrobin_reverse)
+  #     #=> [#<Money value:5.00 currency:USD>, #<Money value:5.01 currency:USD>]
+  def allocate(splits, strategy = :roundrobin)
```

Comment:
this method is getting somewhat complex. Could you add another commit to this PR to move it to a class, something like:
```ruby
def allocate(splits, strategy = :roundrobin)
  Money::Allocate.new(splits, strategy).perform
end
```

Reply:
No problem

Reply:
Done https://github.com/Shopify/money/pull/186/commits/27f09024134d84b35fc6d209e6374cf583375388